pkg/liqo-controller-manager/shadowpod-controller/shadowpod_controller.go

```diff
@@ -59,50 +67,78 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
-	pod := corev1.Pod{
+	// update shadowpod labels to include the "managed-by"
+	shadowPod.SetLabels(labels.Merge(shadowPod.Labels, labels.Set{consts.ManagedByLabelKey: consts.ManagedByShadowPodValue}))
+
+	// if any existing pod is already been created from the shadowpod...
+	existingPod := corev1.Pod{}
+	if err := r.Get(ctx, nsName, &existingPod); err != nil {
+		klog.Error("unable to get pod %q: %v", klog.KObj(&existingPod), err)
```

Comment:
You should immediately return here in case of error.

Reply:
By returning in case of error, the following block of code becomes unreachable since the else block also contains a return statement

Reply:
Sorry, I got confused by your logic. Still, there are two issues here:
* You should immediately return in case any error different from NotFound is raised (as it would be pointless to continue);
* In case the pod is not found, then you should not print an error, as that condition is expected and it will always happen at least once. Doing that would be confusing for anyone reading the logs.